### PR TITLE
[SYCL] Add checking mode in clang-offload-bundler, add FPGA image type triple

### PR DESCRIPTION
**Pull request by domiyan**
1. Add -check-section flag in clang-offload-bunder. Use it to check if type-triple section exists in object file.
2. Add Intel FPGA Image type in triple:
     aoco-intel-linux-sycldevice
     aocr-intel-linux-sycldevice
     aocx-intel-linux-sycldevice
3. Fixed a bug in clang-offload-bundler where in unbundle mode, program may run into dead loop.

Signed-off-by: Domi Yan <domi.yan@intel.com>

### CODE REVIEW

File: clang/include/clang/Driver/Action.h [hunk 1]
```diff
@@ -91,7 +91,12 @@ class Action {
     OFK_Cuda = 0x02,
     OFK_OpenMP = 0x04,
     OFK_HIP = 0x08,
-    OFK_SYCL = 0x10
+    OFK_SYCL = 0x10,
+
```
Comment (kbobrovs) [inline clang/include/clang/Driver/Action.h]: Is it possible that all the FPGA use cases are under OFK_SYCL, and this enum is not changed? FPGA does not really define a new offload kind conceptually - it is still SYCL.

File: clang/lib/Driver/Action.cpp [hunk 2]
```diff
@@ -104,6 +104,12 @@ std::string Action::getOffloadingKindPrefix() const {
     return "device-hip";
   case OFK_SYCL:
     return "device-sycl";
+  case OFK_SYCL_FPGA_IMAGE_AOCO:
+    return "device-fpga-sycl-aoco";
+  case OFK_SYCL_FPGA_IMAGE_AOCR:
+    return "device-fpga-sycl-aocr";
+  case OFK_SYCL_FPGA_IMAGE_AOCX:
+    return "device-fpga-sycl-aocx";
```
Comment (kbobrovs) [inline clang/lib/Driver/Action.cpp]: ... This should probably go away then too

File: clang/lib/Driver/Action.cpp [hunk 3]
```diff
@@ -161,6 +167,12 @@ StringRef Action::GetOffloadKindName(OffloadKind Kind) {
     return "hip";
   case OFK_SYCL:
     return "sycl";
+  case OFK_SYCL_FPGA_IMAGE_AOCO:
+    return "fpga-sycl-aoco";
+  case OFK_SYCL_FPGA_IMAGE_AOCR:
+    return "fpga-sycl-aocr";
+  case OFK_SYCL_FPGA_IMAGE_AOCX:
+    return "fpga-sycl-aocx";
```
Comment (kbobrovs) [inline clang/lib/Driver/Action.cpp]: ... and this

File: clang/lib/Driver/Action.cpp (hunk 2 above)
Comment (domiyan) [reply clang/lib/Driver/Action.cpp]: Reverted

File: clang/lib/Driver/Action.cpp (hunk 3 above)
Comment (domiyan) [reply clang/lib/Driver/Action.cpp]: reverted

File: clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp [hunk 4]
```diff
@@ -84,13 +84,22 @@ static cl::opt<std::string> FilesType(
              "  o   - object\n"
              "  oo  - object; output file is a list of unbundled objects\n"
              "  gch - precompiled-header\n"
-             "  ast - clang AST file"),
+             "  ast - clang AST file\n"
+             "  aoco - unlinked Intel FPGA object file\n"
```
Comment (mdtoguchi) [inline clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp]: Are we taking advantage of -type=aoc*?  All bundle/unbundles being done by the driver use type=o with the appropriate triple.

Comment (domiyan) [reply clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp]: removed.

**Pull request merged**